source/CargoSpace/Assets/Scripts/DirectorBehavior.cs

```diff
@@ -10,65 +10,124 @@
+        if (_npcs.Contains(clicked)) // next: needs to only set target if turret is selected
+        {
+            PlayerTargeted = clicked;
+            //publish 
+            Publish("playerTargetSelected", new Dictionary<string, string>
+            {
+                { "hasTarget", "true" }
+            });
+        }
+        //else if (clicked == <one of the turrets>)
+        //{
+        //embellish to add turrets, check if turret is selected, then go into targeting mode
+        //}
+        else
+        {
+            PlayerTargeted = null;
+            Publish("playerTargetSelected", new Dictionary<string, string>
+            {
+                { "hasTarget", "false" }
+            });
+        }
+    }
+
+    private List<RaycastHit2D> GetRayCastHitList(Vector2 clickLocation)
+    {
+        List<RaycastHit2D> hitResults = new List<RaycastHit2D>();
+        Physics2D.Raycast(clickLocation, Vector2.up * .00001f, new ContactFilter2D(), hitResults);
+        return hitResults;
+    }
+
+    private GameObject GetClickedObject (List<RaycastHit2D> hitResults)
+    {
+        var result = hitResults.FirstOrDefault(hr =>
+            hr.collider &&
+            hr.collider.gameObject &&
+            hr.collider.gameObject.GetComponent<TargetableBehavior>());
+
+        if (!result)
+        {
+            return null;
+        }
+
+        return result.collider.gameObject;
+    }
+
+    private async Task OnTurretFired(Dictionary<string, string> body)
+    {
+        const string key = "source";
+        if (!body.ContainsKey(key))
         {
-            _logger.Bus.LogError(new ArgumentException($"{positionKey} not a valid vector", nameof(arg)));
+            _logger.Bus.LogError(new ArgumentException($"arg does not contain {key}", nameof(body)));
             return;
         }
-        
-        //todo: pass in list, so we know all the things we hit
-        var playerPosition = PlayerShipObject.transform.position;
-        var targetDirection = targetPosition - (Vector2)playerPosition;
+
+        if (!VectorUtils.TryParseVector2(body[key], out var location))
+        {
+            _logger.Bus.LogError(new ArgumentException($"{key} not a valid vector", nameof(body)));
+            return;
+        }
+
+
+        FireLaser(location, (Vector2) PlayerTargeted.transform.position - location);
+    }
+
+
+
+    private void FireLaser(Vector2 source, Vector2 targetDirection)
+    {
         List<RaycastHit2D> hitResults = new List<RaycastHit2D>();
-        Physics2D.Raycast(playerPosition, targetDirection, new ContactFilter2D(),hitResults);
-        var raycastHit2D = hitResults.FirstOrDefault(hr=>
-            hr.collider && 
+        Physics2D.Raycast(source, targetDirection, new ContactFilter2D(), hitResults);
+        var raycastHit2D = hitResults.FirstOrDefault(hr =>
+            hr.collider &&
             hr.collider.gameObject &&
             hr.collider.gameObject.GetComponent<TargetableBehavior>());
```

Comment:
Use GetClickedObject here

Reply:
No, we need the first thing in the ray's path, not just the desired target